src/java/com/puppetlabs/puppetserver/ExecutionStubImpl.java

```diff
@@ -1,8 +1,12 @@
 package com.puppetlabs.puppetserver;
 
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.exec.CommandLine;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 
 public class ExecutionStubImpl {
```

Comment:
Haus mentioned this code could end up extracted into its own library so that code manager can use it as well.  And the clojure side of this code is called `shell-utils`.  Also, it sounded like it could end up doing more than just *executing* shell commands....

So I think I like ShellUtils